File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int KthSmallest(TreeNode root, int k) {
        Stack<TreeNode> stack = new Stack<TreeNode>();
        TreeNode curr = root;
        int count = 0;
        while (curr != null || stack != null) {
            if (curr != null) {
                stack.Push(curr);
                curr = curr.left;
            } else {
                curr = stack.Pop();
                count += 1;
                if (count == k) {
                    return curr.val;
                }
                curr = curr.right;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        stack = [root]
        while stack:
            curr = stack.pop()
            if not curr:
                continue
            
            stack.append(curr.left)
            stack.append(curr.right)

            temp = curr.left
            curr.left = curr.right
            curr.right = temp

        return root